mmv1/third_party/terraform/data_sources/data_source_google_active_folder.go

```diff
@@ -35,26 +35,37 @@ func dataSourceGoogleActiveFolderRead(d *schema.ResourceData, meta interface{})
 		return err
 	}
 
+	var folderMatch *resourceManagerV2.Folder
 	parent := d.Get("parent").(string)
 	displayName := d.Get("display_name").(string)
+	token := ""
 
-	queryString := fmt.Sprintf("lifecycleState=ACTIVE AND parent=%s AND displayName=\"%s\"", parent, displayName)
-	searchRequest := &resourceManagerV2.SearchFoldersRequest{
-		Query: queryString,
-	}
-	searchResponse, err := config.NewResourceManagerV2Client(userAgent).Folders.Search(searchRequest).Do()
-	if err != nil {
-		return handleNotFoundError(err, d, fmt.Sprintf("Folder Not Found : %s", displayName))
-	}
+	for paginate := true; paginate; {
```

Comment:
Do we need to go the pagination route for the data source? For the tests it definitely makes sense, but it seems like the data source could be eventually consistent without it being an issue?

What's the max number of pages this might have to go through?

Reply:
This will not only fix our tests but provide more consistency in behavior from my POV. 

300 folder is usually the limit on folder count. I changed the request to query 300 items at a time.